pkg/setting/setting_test.go

```diff
@@ -427,3 +428,125 @@ func TestGetCDNPathWithAlphaVersion(t *testing.T) {
+				"execute_alerts":             strconv.FormatBool(defaultAlertingExecuteAlerts),
+				"evaluation_timeout_seconds": strconv.FormatInt(defaultAlertingEvaluationTimeoutSec, 10),
+			},
+			verifyCfg: func(t *testing.T, cfg Cfg) {
+				require.Equal(t, 120*time.Second, cfg.AdminConfigPollInterval)
+				require.Equal(t, int64(6), cfg.UnifiedAlertingMaxAttempts)
+				require.Equal(t, int64(60), cfg.UnifiedAlertingMinInterval)
+				require.Equal(t, false, cfg.UnifiedAlertingExecuteAlerts)
+				require.Equal(t, 90*time.Second, cfg.UnifiedAlertingEvaluationTimeout)
+			},
+		},
+		{
+			desc: "when the unified options equal the defaults, it should apply the legacy ones",
+			unifiedAlertingOptions: map[string]string{
+				"admin_config_poll_interval": "120s",
+				"max_attempts":               strconv.FormatInt(defaultAlertingMaxAttempts, 10),
+				"min_interval_seconds":       strconv.FormatInt(defaultUnifiedAlertingMinInterval, 10),
+				"execute_alerts":             strconv.FormatBool(defaultAlertingExecuteAlerts),
+				"evaluation_timeout_seconds": strconv.FormatInt(defaultAlertingEvaluationTimeoutSec, 10),
+			},
+			alertingOptions: map[string]string{
+				"max_attempts":               "12",
+				"min_interval_seconds":       "120",
+				"execute_alerts":             "true",
+				"evaluation_timeout_seconds": "160",
+			},
+			verifyCfg: func(t *testing.T, cfg Cfg) {
+				require.Equal(t, 120*time.Second, cfg.AdminConfigPollInterval)
+				require.Equal(t, int64(12), cfg.UnifiedAlertingMaxAttempts)
+				require.Equal(t, int64(120), cfg.UnifiedAlertingMinInterval)
+				require.Equal(t, true, cfg.UnifiedAlertingExecuteAlerts)
+				require.Equal(t, 160*time.Second, cfg.UnifiedAlertingEvaluationTimeout)
+			},
+		},
+		{
+			desc: "when both unified and legacy options are invalid, apply the defaults",
+			unifiedAlertingOptions: map[string]string{
+				"max_attempts":               "invalid",
+				"min_interval_seconds":       "invalid",
+				"execute_alerts":             "invalid",
+				"evaluation_timeout_seconds": "invalid",
+			},
+			alertingOptions: map[string]string{
+				"max_attempts":               "invalid",
+				"min_interval_seconds":       "invalid",
+				"execute_alerts":             "invalid",
+				"evaluation_timeout_seconds": "invalid",
+			},
+			verifyCfg: func(t *testing.T, cfg Cfg) {
+				require.Equal(t, time.Duration(defaultAdminConfigPollIntervalSec)*time.Second, cfg.AdminConfigPollInterval)
+				require.Equal(t, defaultAlertingMaxAttempts, cfg.UnifiedAlertingMaxAttempts)
+				require.Equal(t, defaultUnifiedAlertingMinInterval, cfg.UnifiedAlertingMinInterval)
+				require.Equal(t, defaultAlertingExecuteAlerts, cfg.UnifiedAlertingExecuteAlerts)
+				require.Equal(t, time.Duration(defaultAlertingEvaluationTimeoutSec)*time.Second, cfg.UnifiedAlertingEvaluationTimeout)
+			},
+		},
+		{
+			desc: "when unified alerting options are invalid, apply legacy options",
```

Comment:
My impression it that we don't usually error if a configuration is invalid: we just apply a default. I believe that this is a deliberate choice so that the server does not fail to start in case of invalid configuration. @grafana/backend-platform WDYT?

Reply:
If that's the case then consider this resolved.